jane/doc/extensions/stack/reference.md

```diff
@@ -179,10 +180,25 @@ value `outer`, while being local, was definitely not local to the region of
 compiler know that it is safe to still refer to `outer` from within the closure
 `g`? See "Closures" below for more details)
 
+This situation also arises with local parameters. For example:
+
+```ocaml
+let f (local_ x) =
+  let local_ y = 3 :: x in
+  ??
+```
+
+Both `x` and `y` are local and cannot, in general, escape a region. However,
+filling `??` in with `x` (but not `y`) is allowed. This is because we know that
+the value pointed to by `x` must have existed before `f` started and thus is
+guaranteed to be allocated in a pre-existing stack frame (or on the heap).
+In contrast, `y` will point to a cons cell allocated in the current stack frame;
+returning `y` from `f` would be unsafe (and is a type error).
+
 ## Inference
 
 In fact, the allocations of the examples above will be on
-stack even without the `stack_` keyword, if it is safe to do
+stack even without the `stack_` or `local_` keywords, if it is safe to do
```

Comment:
(similarly, avoid using local_ for stack allocation)

Reply:
As above, I don’t think we can avoid `local_` in stack-allocation programming. I *do* think we should avoid saying things like “stuff you label as `local` gets allocated on the stack”, because the reality is considerably more subtle. But that doesn’t mean `local_` is useless!

Reply:
I continue to think it's better with `local_` there, because the last example above uses `local_`.

Reply:
Yeah but that `local_` is used strictly for escaping analysis, not to do with stack allocation (even though the description still mentions stack allocation).

Reply:
Ahh good catch. It should say "allocations in general".